shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -371,7 +394,10 @@ class JSBackend {
           .concat(otherStmts.flatMap {
             case Def(isRecursive, name, L(body)) =>
               val translatedBody = translateTerm(body)
-              val tempName = getTemporaryName(name)
+              // Get a name that not conflicts with class names.
+              val tempName = (name #:: LazyList
+                .from(0)
```

Comment:
Does this really work with defs that have the same name? Looks like you're going to start from 0 every time.

Reply:
Oh you're right! I’m fixing it soon.